crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -0,0 +1,316 @@
+use crate::{
+    environment::EnvPtr,
+    error::{mdbx_result, Result},
+    CommitLatency,
+};
+use std::{
+    ptr,
+    sync::mpsc::{sync_channel, Receiver, SyncSender},
+};
+
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct TxnPtr(pub(crate) *mut ffi::MDBX_txn);
+unsafe impl Send for TxnPtr {}
+unsafe impl Sync for TxnPtr {}
+
+pub(crate) enum TxnManagerMessage {
+    Begin { parent: TxnPtr, flags: ffi::MDBX_txn_flags_t, sender: SyncSender<Result<TxnPtr>> },
+    Abort { tx: TxnPtr, sender: SyncSender<Result<bool>> },
+    Commit { tx: TxnPtr, sender: SyncSender<Result<(bool, CommitLatency)>> },
+}
+
+#[derive(Debug)]
+pub(crate) struct TxnManager {
+    sender: SyncSender<TxnManagerMessage>,
+    #[cfg(feature = "read-tx-timeouts")]
+    read_transactions: Option<std::sync::Arc<read_transactions::ReadTransactions>>,
+}
+
+impl TxnManager {
+    pub(crate) fn new(env: EnvPtr) -> Self {
+        let (tx, rx) = sync_channel(0);
+        let txn_manager = Self {
+            sender: tx,
+            #[cfg(feature = "read-tx-timeouts")]
+            read_transactions: None,
+        };
+
+        txn_manager.start_message_listener(env, rx);
+
+        txn_manager
+    }
+
+    fn start_message_listener(&self, env: EnvPtr, rx: Receiver<TxnManagerMessage>) {
```

Comment:
I need some docs here

and this should also cover what initially tracks open transactions